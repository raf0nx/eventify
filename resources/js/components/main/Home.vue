<template>
	<v-main>
		<div class="container-fluid" style="height: 1500px">
			<div class="display-1">Home</div>
			<v-btn @click="logout()" data-cy="logout_button">Logout</v-btn>
			<v-btn @click="getUser()">Get User</v-btn>
		</div>
	</v-main>
</template>

<script lang="ts">
	import { Vue, Component } from "vue-property-decorator";

	import AuthService from "@/services/AuthService";
	import { AuthModule } from "@modules/Auth";

	@Component({})
	export default class Home extends Vue {
		logout(): void {
			AuthModule.logout();
		}

		async getUser(): Promise<void> {
			const user = await AuthService.getAuthUser();
			console.log(user);
		}
	}
</script>

<style>
</style>