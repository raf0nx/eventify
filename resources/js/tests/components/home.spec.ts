import { createLocalVue, shallowMount, Wrapper } from "@vue/test-utils";
import Vuetify from "vuetify";

import Home from "@components/main/Home.vue";

describe("Home.vue", () => {
    const localVue = createLocalVue();
    let wrapper: Wrapper<Home>;
    let vuetify: Vuetify;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = shallowMount(Home, { localVue, vuetify });
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it("Should match snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Should trigger logout method", async () => {
        const logoutSpy = jest
            // @ts-ignore
            .spyOn(wrapper.vm, "logout")
            .mockResolvedValue(true);
        // @ts-ignore
        const isLoggedOut = await wrapper.vm.logout();

        expect(logoutSpy).toBeCalled();
        expect(isLoggedOut).toBe(true);
    });
});
