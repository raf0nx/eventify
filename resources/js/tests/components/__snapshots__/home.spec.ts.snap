// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue Should match snapshot 1`] = `
"<v-main-stub tag=\\"main\\">
  <div class=\\"container-fluid\\" style=\\"height: 1500px;\\">
    <div class=\\"display-1\\">Home</div>
    <v-btn-stub tag=\\"button\\" activeclass=\\"\\" type=\\"button\\" id=\\"test\\" data-cy=\\"logout_button\\">Logout</v-btn-stub>
    <v-btn-stub tag=\\"button\\" activeclass=\\"\\" type=\\"button\\">Get User</v-btn-stub>
  </div>
</v-main-stub>"
`;
